### PR TITLE
Refactor module p

### PR DESCRIPTION
**Pull request by epatrizio**
Preparing work for fuzzing (done with @chambart)

### CODE REVIEW

File: src/symbolic.ml [hunk 1]
```diff
@@ -33,7 +55,7 @@ module P = struct
 
   type thread = Thread.t
 
-  module Choice = Symbolic_choice.MT
+  module Choice = Choice_monad
```
Comment (zapashcanon) [inline src/symbolic.ml:58]: This can be removed if you rename the functor parameter `Choice_monad` to `Choice`.

File: src/symbolic.ml [hunk 2]
```diff
@@ -6,7 +6,29 @@ open Types
 module Def_value = Concrete_value
 module Solver = Thread.Solver
 
-module P = struct
+module type Thread = sig
+  type t
+
+  type 'a solver_module = (module Encoding.Solver_intf.S with type t = 'a)
+  type solver = S : 'a solver_module * 'a -> solver
```
Comment (krtab) [inline src/symbolic.ml:13]: This and

```ocaml
  val solver: t -> solver

  val pc: t -> Symbolic_value.S.vbool list
```

were added unsupervised by me and @epatrizio so you may want to use double the amount of caution when reviewing.

Comment (zapashcanon) [reply src/symbolic.ml:13]: I'm not sure this is what we want, but you'll find out while writing the fuzzer part. :-)

**Comment by krtab** [comment]
Can you run `dune fmt` ?

**Review by krtab — APPROVED** [review]

**Pull request merged**